### ISSUE TITLE
Native compilation broken by adding "queryAllDeclaredMethods": true to Java records

### ISSUE THREAD

**Issue by sdeleuze** [oracle/graal#6510]
It looks like we have uncovered a potential GraalVM bug that breaks Spring Boot 3.0.6+ application using Spring beans that are records, which is a pretty propular use cases with `@ConfigurationProperties`. The related Spring Framework issue is https://github.com/spring-projects/spring-framework/issues/30383, and it looks like a record specific follow-up of https://github.com/oracle/graal/issues/5139 cc @cstancu @loicottet.

This regression has been cause by an IMO [legit Spring Framework refinement](https://github.com/spring-projects/spring-framework/commit/b3748243199fe809f177bb7922c69f7eb37ab6d2) that adds `"queryAllDeclaredMethods": true` on all beans for 2 reasons:
 - It is by design required if we want properly to support init/destroy methods
 - It will improve the compatiblity with upcoming GraalVM reflection support changes

So this works:
```
{
    "name": "com.example.SampleConfigurationProperties",
    "methods": [
      {
        "name": "<init>",
        "parameterTypes": [
          "java.lang.String"
        ]
      }
    ]
  }
```

And this fails:
```
{
    "name": "com.example.SampleConfigurationProperties",
    "queryAllDeclaredMethods": true,
    "methods": [
      {
        "name": "<init>",
        "parameterTypes": [
          "java.lang.String"
        ]
      }
    ]
  }
```
With the following error:
```
Fatal error: com.oracle.svm.core.util.VMError$HostedError: com.oracle.svm.core.util.VMError$HostedError: New Method or Constructor found as reachable after static analysis: public java.lang.String com.example.SampleConfigurationProperties.message()
```
Can be reproduced with https://github.com/sdeleuze/demo-profile-aot when updating to Boot 3.0.6.

As part of https://github.com/spring-projects/spring-framework/issues/30383 and since this is a blocking issue, we will likely workaround by adding invocation metadata which works:
```
{
    "name": "com.example.SampleConfigurationProperties",
    "queryAllDeclaredMethods": true,
    "allDeclaredMethods": true,
    "methods": [
      {
        "name": "<init>",
        "parameterTypes": [
          "java.lang.String"
        ]
      }
    ]
  }
```

Could you please fix this and backport it to GraalVM 22.3.x maintenance release?

**Comment by christianwimmer** [comment]
I'll have a look at the problem

**Comment by christianwimmer** [comment]
The bug is already fixed by #5643 in master and the upcoming 23.0 release. I will make sure the fix gets backported to the next 22.3 update release.
